File: backend/app/Models/Invoice.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Invoice extends Model
{
    protected $fillable = [
        'sales_order_id',
        'invoice_date',
        'due_date',
        'total_amount',
        'status',
        'company_id',
    ];

    protected $casts = [
        'invoice_date' => 'date',
        'due_date' => 'date',
        'total_amount' => 'decimal:2',
    ];

    public function salesOrder(): BelongsTo
    {
        return $this->belongsTo(SalesOrder::class);
    }

    public function company(): BelongsTo
    {
        return $this->belongsTo(Company::class);
    }
}

File: lib/features/sales/domain/entities/invoice.dart
class Invoice {
  final int id;
  final int salesOrderId;
  final DateTime invoiceDate;
  final DateTime? dueDate;
  final double totalAmount;
  final String status;

  Invoice({
    required this.id,
    required this.salesOrderId,
    required this.invoiceDate,
    this.dueDate,
    required this.totalAmount,
    required this.status,
  });
}
